purchase_split_wizard_deposit/wizards/purchase_split_wizard.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2024 Alberto Martínez <alberto.martinez@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, exceptions, fields, models
+
+
+class PurchaseSplitWizard(models.TransientModel):
+    _inherit = "purchase.split.wizard"
+
+    split_advances = fields.Boolean()
+
+    advance_line_ids = fields.One2many(
+        string="Advance lines",
+        comodel_name="purchase.advance.line.split.wizard",
+        inverse_name="wizard_id",
+    )
+
+    def get_default_lines(self, purchase_id):
+        old_res = super().get_default_lines(purchase_id)
+        advance_line_ids = purchase_id.order_line.filtered("is_deposit").mapped("id")
+        new_res = []
+        for command in old_res:
+            if command[2].get("purchase_line_origin_id") not in advance_line_ids:
+                new_res.append(command)
+        return new_res
+
+    def default_get(self, fields_list):
+        res = super().default_get(fields_list)
+        purchase_id = self.env["purchase.order"].browse(
+            res.get("purchase_origin_id")
+            or self.env.context.get("default_purchase_origin_id")
+        )
+        splitteable_advance_lines = purchase_id.order_line.filtered(
+            lambda li: ((li.qty_splittable or li.display_type) and li.is_deposit)
+        )
+        res.update(
+            {
+                "split_advances": any(splitteable_advance_lines),
```

Comment:
Using just splitteable_advance_lines doesn't work?

Reply:
It does work, but the any converts it to boolean and i like it :3